### PR TITLE
CP-27709: suppress error if blkdiscard fails on full provisioned lun

### PR DESCRIPTION
**Pull request by MarkSymsCtx**
Signed-off-by: Mark Syms <mark.syms@citrix.com>

### CODE REVIEW

File: drivers/scsiutil.py [hunk 1]
```diff
@@ -182,9 +182,8 @@ def scsi_dev_ctrl(ids, cmd):
             opterr='An error occured during the scsi operation')
 
 def getdev(path):
-    realpath = os.path.realpath(path)
```
Comment (MarkSymsCtx) [inline drivers/scsiutil.py]: It turns out that calling realpath here meant that the match_dm check in the next line could never be true as any device of the form `/dev/mapper/xxxxx/` will realpath to something of the form `/dev/dm-xx`

**Comment by MarkSymsCtx** [comment]
It now just logs to SMlog if a discard to a full provisioned LUN fails (in most cases we'd expect that it will but there may be some SANs out there that will accept a discard on a LUN marked full).

**Comment by stefanopanella** [comment]
Lgtm

File: drivers/lvutil.py [hunk 2]
```diff
@@ -725,3 +728,23 @@ def removeDevMapperEntry(path, strict=True):
         util.SMlog("removeDevMapperEntry: dmsetup remove failed for file %s " \
                    "with error %s, and lsof ret is %s." % (path, str(e), ret))
         return False
+
+def vgIsFullyProvisioned(vgname):
+    """
+    Check the PVs behind the specified VG and return
+    True if any of them is fully provisioned
+    """
+    text = cmd_lvm([CMD_VGDISPLAY, '-v', vgname])
```
Comment (chandrikas) [inline drivers/lvutil.py]: Is there an easier way to find underlying devices? Can we use self.root?

Comment (chandrikas) [reply drivers/lvutil.py]: hmm, I see that getting from lvutil to LVHDSR is going to be convoluted too. Let's go with this approach then.

**Pull request merged**